api/src/main/java/org/allaymc/api/blockentity/component/BlockEntityBaseComponent.java

```diff
@@ -18,7 +19,15 @@
 /**
  * @author daoge_cmd
  */
-public interface BlockEntityBaseComponent extends BlockEntityComponent {
+public interface BlockEntityBaseComponent extends BlockEntityComponent, PersistentDataHolder {
+    String TAG_ID = "id";
```

Comment:
I prefer keeping these tags in correspond implementations, as plugins are not expected to touch the encoded nbts directly